MSOLSpray.py

```diff
@@ -70,6 +72,9 @@
 
     r = requests.post(f"{url}/common/oauth2/token", headers=headers, data=body)
 
+    if sleep is not None:
+        time.sleep(int(sleep))
```

Comment:
If you change the argparse argument to default to 0 as suggested, there will not be any need for this `is not None` check.

In addition, I would suggest moving the `time.sleep(sleep)` to the end of the `for username in usernames:`-loop.  Currently your code will sleep between the request and the parsing of the request. I think it makes more sense to put it after the parsing, right before starting a new round in the loop.

Reply:
ok, i think i've sorted this. I moved the sleep to the start of the loop, except on first go, otherwise I need to catch the "continue" statements as well.